src/plugins/visualizations/public/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -118,9 +137,18 @@ export class VisualizeEmbeddableFactory
   public async create() {
     // TODO: This is a bit of a hack to preserve the original functionality. Ideally we will clean this up
     // to allow for in place creation of visualizations without having to navigate away to a new URL.
+    let originatingAppParam = this.currentAppId;
```

Comment:
Instead of subscribing in constructor and having `. unsubscribeSubscriptions()` method, could this line be something like

```suggestion
    let originatingAppParam = await this.getCurrentAppId();
```

Reply:
Or may be inlining the pipe trick

```ts
let originatingAppParam;
core.application.currentAppId$
  .pipe(take(1))
  .subscribe(appId => originatingAppParam = appId);
```

Reply:
I like @streamich 's suggestion here for using `await` instead of a subscription ☝️

Reply:
Implemented this suggestion!